src/hera/workflows/v5/dag.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from typing import Any, List, Optional
+
+from hera.workflows.models import DAGTask
+from hera.workflows.models import DAGTemplate as _ModelDAGTemplate
+from hera.workflows.models import Template as _ModelTemplate
+from hera.workflows.v5._mixins import _DAGTaskMixin, _IOMixin, _TemplateMixin
+
+
+class DAG(_IOMixin, _TemplateMixin):
+    name: str
+    fail_fast: Optional[bool] = None
+    target: Optional[str] = None
+    tasks: List[DAGTask] = []
+    _task_queue: List[_DAGTaskMixin] = []
+
+    def __enter__(self) -> DAG:
+        """Enter the context of the workflow"""
+        from hera.workflows.v5._context import _context
+
+        _context.enter(self)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
+        from hera.workflows.v5._context import _context
+
+        for task in self._task_queue:
+            self.tasks.append(task._build_dag_task())
```

Comment:
I believe we should be able to simplify this if we inherit from _SubNodeMixin and get rid of these two methods?